test/valid/test0198.d

```diff
@@ -0,0 +1,86 @@
+//T compiles:yes
+//T has-passed:yes
+//T retval:0
+// GC slab finalizers
+
+extern(C) void __sd_gc_collect();
+extern(C) void* __sd_gc_alloc_finalizer(size_t size, void* finalizer);
+extern(C) void* __sd_gc_tl_flush_cache();
+
+static int destructorSum;
+struct SlabDestructor(size_t size) {
+	// The -1 is for the metadata storage.
+	size_t[size / size_t.sizeof - 1] x;
+
+	~this() {
+		destructorSum += x[0];
+	}
+}
+
+void destroyItem(T)(void* item, size_t size) {
+	assert(size == T.sizeof, "Incorrect size passed to destructor.");
+	(cast(T*) item).__dtor();
+}
+
+void allocateItem(size_t sz)() {
+	alias T = SlabDestructor!(sz);
+	auto item = cast(T*) __sd_gc_alloc_finalizer(T.sizeof, &destroyItem!T);
+	item.x[0] = sz;
+}
+
+void allocateItems() {
+	allocateItem!(16)();
+	allocateItem!(24)();
+	allocateItem!(32)();
+	allocateItem!(40)();
+	allocateItem!(48)();
+	allocateItem!(56)();
+	allocateItem!(64)();
+	allocateItem!(80)();
+	allocateItem!(96)();
+	allocateItem!(112)();
+	allocateItem!(128)();
+	allocateItem!(160)();
+	allocateItem!(192)();
+	allocateItem!(224)();
+	allocateItem!(256)();
+	allocateItem!(320)();
+	allocateItem!(384)();
+	allocateItem!(448)();
+	allocateItem!(512)();
+	allocateItem!(640)();
+	allocateItem!(768)();
+	allocateItem!(896)();
+	allocateItem!(1024)();
+	allocateItem!(1280)();
+	allocateItem!(1536)();
+	allocateItem!(1792)();
+	allocateItem!(2048)();
+	allocateItem!(2560)();
+	allocateItem!(3072)();
+	allocateItem!(3584)();
+	allocateItem!(4096)();
+	allocateItem!(5120)();
+	allocateItem!(6144)();
+	allocateItem!(7168)();
+	allocateItem!(8192)();
+	allocateItem!(10240)();
+	allocateItem!(12288)();
+	allocateItem!(14336)();
+}
```

Comment:
FYI, I wanted to use some sort of static looping, but I don't think sdc supports it, let me know if you can think of a way to do this...

Reply:
A recursive template could do, but I'm not sure this is necessary.